### ISSUE TITLE
Overfitting

### ISSUE THREAD

**Issue by rshanor** [alexgkendall/SegNet-Tutorial#29]
Hey all,
I followed the tutorial trying to use Segnet with my own dataset. I only have about 90 labeled images at the moment, although could capture more if necessary. In my dataset, there are only 2 labeled classes. I am running into problems overfitting; accuracy approaches 97% on my training data but is significantly worse during testing. I am currently working to artificially generate more training data by applying transforms to my dataset. Does anyone have other advice or experience reducing overfitting using the segnet architecture?  
Thanks,
Rick

**Comment by nshaud** [comment]
From my own experience, you can fight this overfitting generating new images by applying random crops, rotations, symetries etc. You may also want to try dropout (e.g. in the Bayesian SegNet model), but I'd first consider augmenting the dataset to see if it helps.

**Comment by rshanor** [comment]
Yes so far augmenting the dataset has worked pretty well! Gotten from about 90% accuracy on 2 class label to 95%
